tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/query/QueryActionOrExprTest.java

```diff
@@ -141,6 +145,65 @@ public void testQueryActionOrExprSemanticsNegative() {
         validateError(negativeResult, i++, "action invocation as an expression not allowed here", 315, 23);
         validateError(negativeResult, i++, "incompatible types: expected 'int', found 'other'", 316, 21);
         validateError(negativeResult, i++, "action invocation as an expression not allowed here", 320, 50);
+        validateError(negativeResult, i++, "incompatible types: expected '(T3[]|T4[])', found '(T3|T4)[]'",
+                339, 13);
+        validateError(negativeResult, i++, "incompatible types: expected 'T3', " +
```

Comment:
Why we should expect this error?
If we consider a normal ternary expression this is allowed
```ballerina
public function foo(string str) {
    T3|T4 _ = str == "" ? <T3>{
            str: ""
        } : <T4>{
            foo: false
        };
}

```

Reply:
In a normal expression this is correct. But In here we log the error inside the select clause. In the L.H.S it is expecting a `T3[]|T4[]`.  So select clause type being a  `T3|T4` is incorrect. 

Initially we agreed to log a single error for the whole query. But when reviewing this PR  @MaryamZi pointed out a [  #case](https://github.com/ballerina-platform/ballerina-lang/pull/40835#discussion_r1259607967) where we needed type check the select clause as well.